### ISSUE TITLE
Promote FStar from Homebrew/Science

### ISSUE THREAD

**Issue by beurdouche** [Homebrew/homebrew#44797]
@DomT4, @dpo, @bfontaine, @dunn 
Hello Hello !
Last week, we decided to move FStar from the Core to the Science tap (#44341) because it was dependant on Z3. =)
Now Z3 was promoted to Core #44652 and all similar programming languages like Rust, Coq are in Core.

Do you mind if we move FStar back to Core ? =)

**Comment by DomT4** [comment]
If you can get @Homebrew/science and another one/two of @Homebrew/owners to sign off on it, yes, feel free. No objections personally. You'd need to wait on the `z3` formula to be merged here, I'd presume.

**Comment by dpo** [comment]
Apparently
* `z3` has `fstar` as unique dependent
* `fstar` has no dependent
* `cvc4` has `cryptol` as unique dependent, which is a cryptography package
* `cryptol` has no dependent.

So `z3 -> fstar` and `cvc4 -> cryptol` are two pretty small dependency graphs.

Whether `cryptol` is a "scientific" package is debatable but the home page says

> A Cryptol implementation of an algorithm resembles its mathematical specification more closely than an implementation in a general purpose language.

So either way, they should probably all be in core, or all in science.

On a side note:
```
$ brew deps fstar
Error: No available formula for fsharp 
```

**Comment by beurdouche** [comment]
@Dpo : thanks, it is fixed already locally and ready to push as soon as a decision is reached here. =)

**Comment by bfontaine** [comment]
I don’t have any hard objection; but why do you want to move formulae from homebrew/science to the core?

**Comment by beurdouche** [comment]
I submitted it in core in the first place because I believe that such a programming language is a more general usage than a domain specific tools. I do not hesitate to submit scientific tools to `science` anyway (see `proverif`, `cryptoverif`, soon `flextls`) but I don't believe FStar should be in `science`, for this reason and because of it's increasing use. I am not opposed to F* staying in `science` but I would really prefer having it where similar languages and tools are ie. in core.

**Comment by bfontaine** [comment]
Thank you for the explanation. I’m :+1: here.

**Comment by beurdouche** [comment]
Should I proceed ?

**Comment by bfontaine** [comment]
Yep.

**Comment by DomT4** [comment]
Closing in favour of #44989.
